crates/cli/src/subcommands/generate/csharp.rs

```diff
@@ -1031,6 +1048,13 @@ fn autogen_csharp_access_funcs_for_struct(
                             col_i
                         )
                         .unwrap();
+                    } else if is_option {
+                        writeln!(
+                            output,
+                            "var compareValue = ({})(productValue.elements[{}].AsSumValue().tag == 1 ? null : productValue.elements[{}].AsSumValue().value.As{}());",
```

Comment:
Is possible to get the `optional` by the user? If that is the case then in C# 8 exists a proper `optional` type, or else adding a little help that constructs it. If this is only used internally then ok.